Overhaul tags and tag banning (#10941)
smogon/pokemon-showdown: sim/NONSTANDARD.md

- item: Berserk Gene (only exists in Gen 2 games)
- move: Hidden Power (only exists in Gen 2-7 games)

These are tagged `Nonexistent`, and also `Past`, `Future`, or `LGPE`, depending on the game.

Things that only existed in Gen 5 and earlier no longer exist in future games' game data at all. In those future gens, these are tagged `Past` and `True Past`.

> Things that existed in Gen 6 or later and then stopped existing are "Dexited". In those future gens, they partially exist in game data (frequently, they will have stats but not graphics), and attempting to hack them in will be buggy. These will only be tagged `Past`.

Author: ```suggestion
Things that existed in Gen 7 (other than Totems) and then stopped existing in Gen 8 or later are "Dexited". In those future gens, they partially exist in game data (frequently, they will have stats but not graphics), and attempting to hack them in will be buggy. These will only be tagged `Past`.
```
I also got the Dexit date wrong.

The data contains information for moves named "Hidden Power Fire", "Hidden Power Ice", etc. These are provided for convenience: when building teams (to set the move to "Hidden Power" and change IVs), when running damage calculations, and when exporting or sending movesets.

These are matched with `move.placeholderFor === "Hidden Power"`. Since no other move has this feature, you can also check `move.placeholderFor` for existence.

Max moves and G-Max moves
-------------------------

> Max moves cannot be learned normally, and behave weirdly (have 0 base power, dealing exactly 2 damage) if hacked into a Pokémon's moveset. G-Max moves are similar, but have normal base power.

Author: ```suggestion
Max and G-Max moves cannot be learned normally, and behave weirdly (sometimes have 0 base power, dealing exactly 2 damage) if hacked into a Pokémon's moveset.
```
